src/k8s/pkg/k8sd/app/app.go

```diff
@@ -242,3 +233,34 @@ func (a *App) markNodeReady(ctx context.Context, s state.State) {
 		}
 	}
 }
+
+func (a *App) waitKubernetesReady(ctx context.Context, s state.State) error {
+	dbOpen := make(chan struct{})
+	go a.waitForDBOpen(ctx, s, dbOpen)
+
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("failed to wait for db to be open: %w", ctx.Err())
+	case <-dbOpen:
+	}
+
+	logger := log.FromContext(ctx)
+
+	// also check kubernetes endpoint
+	if err := control.WaitUntilReady(ctx, func() (bool, error) {
+		client, err := a.snap.KubernetesClient("")
```

Comment:
caredul, this creates an admin client, which will fail on worker nodes

perhaps you can use a node client instead

Reply:
right, and I guess we should get the config for csrsigning controller from a node client as well?